geopandas/_vectorized.py

```diff
@@ -728,6 +729,28 @@ def almost_equals(self, other, decimal):
 #
 
 
+def clip_by_rect(data, xmin, ymin, xmax, ymax):
+    if compat.USE_PYGEOS:
+        return pygeos.clip_by_rect(data, xmin, ymin, xmax, ymax)
+    else:
+        if not compat.SHAPELY_GE_17:
+            raise ValueError(
+                "Fast clipping by rectangle is not available in your shapely version. "
+                "It was introduced in shapely 1.7. Please update your version of "
+                "shapely or use geopandas with pygeos (https://geopandas.org/"
+                "getting_started/install.html#using-the-optional-pygeos-dependency)"
+            )
```

Comment:
Assuming #2358 will be merged, we will require shapely 1.7 and this can be cleared. The CI will complain until then though.
```suggestion
```